lib/worker.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const { isMainThread, parentPort, workerData } = require('worker_threads')
+const track = require('./progressTracker')
+const run = require('./run')
+
+function runTracker (argv, ondone) {
+  const tracker = run(argv)
+
+  tracker.on('done', (result) => {
+    if (ondone) ondone(result)
+  })
+
+  tracker.on('tick', (data) => {
+    parentPort.postMessage({ cmd: 'TICK', data })
+  })
+
+  tracker.on('error', (err) => {
+    if (err) {
+      throw err
+    }
+  })
+
+  // if not rendering json, or if std isn't a tty, track progress
+  if (!argv.json || !process.stdout.isTTY) track(tracker, argv)
```

Comment:
I don't understand the isTTY part, can you clarify?

Reply:
This was pre-existing in [autocannon.js](https://github.com/mcollina/autocannon/pull/290/files#diff-d19ffc6d9071d9610442b1d9173acd4c390dd5001bdc732bee7cd75e5f96aa36L260) when I used the same in worker.js. But it doesn't sound right to be in here. Removed.